hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -172,6 +173,16 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
         throw new OMException("Failed to commit key, as " + dbOpenKey +
             "entry is not found in the OpenKey table", KEY_NOT_FOUND);
       }
+
+      // Ensure the parent exist.
+      if (!"".equals(OzoneFSUtils.getParent(keyName))
```

Comment:
Can you move this check inside `ozoneManager.getEnableFileSystemPaths() == true` case to do this only for FileSystem api. 

Should we need this check for bucket#createKey & OzoneOutputStream#close() in turn make OMKeyCommitRequest call?

Reply:
Thanx @rakeshadr for the review.
I have moved the check inside `ozoneManager.getEnableFileSystemPaths() == true` and added the same logic and tests for `OMAllocateBlockRequest` as well. Please have a look..